core/vm/contracts_suave_eth.go

```diff
@@ -43,28 +41,39 @@ var (
 	fillMevShareBundleAddress  = common.HexToAddress("0x43200001")
 )
 
-type simulateBundle struct {
-}
+func signEthTransactionImpl(txn []byte, chainId string, signingKey string) ([]byte, error) {
+	key, err := crypto.HexToECDSA(signingKey)
+	if err != nil {
+		return nil, fmt.Errorf("key not formatted properly: %w", err)
+	}
 
-func (c *simulateBundle) RequiredGas(input []byte) uint64 {
-	// Should be proportional to bundle gas limit
-	return 10000
-}
+	chainIdInt, err := hexutil.DecodeBig(chainId)
```

Comment:
should this be fully generic or only support specific chainIDs for now? just so people dont shoot themselves in the foot since we only support eth L1 rn

Reply:
:shrug: